### PR TITLE
Added to updates 1 and 2 - carpedata

### PR DESCRIPTION
**Pull request by ssheridanucsb**

### CODE REVIEW

File: ucsb_ds_capstone_projects_2021/projects/carpedata/update1.md [hunk 1]
```diff
@@ -1,15 +1,52 @@
...
+    words = [word.lower() for word in tokens if word.isalpha() and word not in stop_words]
+    w = " ".join(words)
+    return w
+  ```
+  
+### Logistic Regression 
+ 
+ The classification report of our logistic regression showed a promissing model. 
+ 
+<img width="483" alt="Screen Shot 2021-03-15 at 5 29 31 PM" src="https://user-images.githubusercontent.com/43657731/111238284-04702d00-85b4-11eb-9027-c6fa7d20a767.png">
```
Comment (syoh) [inline ucsb_ds_capstone_projects_2021/projects/carpedata/update1.md]: Markdown can embed images directly: https://jupyterbook.org/content/figures.html
For the image `someimg.png` in the same directory as the markdown file, replace the HTML with `some-text`

File: ucsb_ds_capstone_projects_2021/projects/carpedata/update1.md [hunk 2]
```diff
@@ -1,15 +1,52 @@
...
+
+Examing the words associated with the largest coefficients of our model showed some seemingly intuitive results.
+ 
+<img width="672" alt="Screen Shot 2021-03-15 at 5 30 22 PM" src="https://user-images.githubusercontent.com/43657731/111238348-22d62880-85b4-11eb-83c4-1dd7e1fbf8da.png">
```
Comment (syoh) [inline ucsb_ds_capstone_projects_2021/projects/carpedata/update1.md]: Same here
Markdown can embed images directly: https://jupyterbook.org/content/figures.html
For the image `someimg.png` in the same directory as the markdown file, replace the HTML with `some-text`

File: ucsb_ds_capstone_projects_2021/projects/carpedata/update1.md [hunk 3]
```diff
@@ -1,15 +1,52 @@
...
+
+The precision-recall curve shows our model needs very little hyperparameter tuning for our desired goal of maximizing recall. 
+
+<img width="415" alt="Screen Shot 2021-03-15 at 5 32 56 PM" src="https://user-images.githubusercontent.com/43657731/111238517-7e081b00-85b4-11eb-862d-30c9d115bf56.png">
```
Comment (syoh) [inline ucsb_ds_capstone_projects_2021/projects/carpedata/update1.md]: Same here
Markdown can embed images directly: https://jupyterbook.org/content/figures.html
For the image `someimg.png` in the same directory as the markdown file, replace the HTML with `some-text`

File: ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md [hunk 4]
```diff
@@ -1,11 +1,52 @@
...
+
+## Model Progress
+
+Since our last update we have experimented with a variety of new models. We implemented gradient-boosted-trees using LightGBM and tuned our models hyperparameters using Optuna. Gradient-boosted trees are better for highly unbalanced data like ours where the minority class (positive labels--the ones containing criminal record information) hold greater value than the majority (negative) class and we care more about false negatives than false positives. After tuning we were able to achieve Recall of 96%. A major improvement over our original random forest model.
+
+![image2](https://user-images.githubusercontent.com/43657731/111241637-c7f3ff80-85ba-11eb-846f-d41156020231.png)
```
Comment (syoh) [inline ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md]: Place the image in the same directory. Here the link doesn't show which repository the image is coming from

File: ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md [hunk 5]
```diff
@@ -1,11 +1,52 @@
...
+
+In order to interpret the model, we plotted variable importance using lightGBM’s feature_importances method. We found the 25 most important features in our model.
+
+![image4](https://user-images.githubusercontent.com/43657731/111241691-dd692980-85ba-11eb-868d-e0ddcf0c3df6.png)
+
+As shown in the plot, the words ‘arrested’, ‘charged’, ‘police’ and ‘crime’ are the most important and are words we would associate with criminal records. 
+
+## Decision Threshold Tuning
+
+![image1](https://user-images.githubusercontent.com/43657731/111241726-f1149000-85ba-11eb-97b9-c71dc8172a1c.png)
```
Comment (syoh) [inline ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md]: Same here
Place the image in the same directory. Here the link doesn't show which repository the image is coming from

File: ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md [hunk 6]
```diff
@@ -1,11 +1,52 @@
...
+
+A probability threshold of ~0.65 would optimize both recall and precision, however we opted to use a threshold of 0.2 since recall is more important to us. A threshold of 0.2 maximizes recall without sacrificing too much precision and accuracy
+
+### Hyperparameter Tuning
+
+Optuna is a python package that runs trials by case on a designated objective function. For hyperparameter tuning, the objective function consists of ranges of parameters tested and tests for best parameters values in that range. Optuna was found to be more efficient and faster than GridSearchCVParameter output is also more precise given its range, not from manually inputted values. For the above plot, over the number of trials, the search objective value becomes closer to 1 or -1,  indicating strong model correlation to the dataset.
+
+![image5](https://user-images.githubusercontent.com/43657731/111241943-641e0680-85bb-11eb-87f2-84083acd8e8d.png)
```
Comment (syoh) [inline ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md]: Place the image in the same directory. Here the link doesn't show which repository the image is coming from

File: ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md [hunk 7]
```diff
@@ -1,11 +1,52 @@
...
+
+
+
+## Model Assessment
+
+![image3](https://user-images.githubusercontent.com/43657731/111242010-831c9880-85bb-11eb-932c-cd545e251070.png)
```
Comment (syoh) [inline ucsb_ds_capstone_projects_2021/projects/carpedata/update2.md]: Place the image in the same directory. Here the link doesn't show which repository the image is coming from

**Review by syoh — CHANGES_REQUESTED** [review]

**Review by syoh — APPROVED** [review]
looks good
